google-cloud-spanner/src/main/java/com/google/cloud/spanner/StreamingResultSet.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.spanner;
+
+/** Streaming implementation of ResultSet that supports streaming of chunks */
+public interface StreamingResultSet extends ResultSet {
+
+  /**
+   * Returns the {@link boolean} for this {@link ResultSet}. This method will be used by
+   * AsyncResultSet to initiate gRPC streaming
+   */
+  boolean initiateStreaming(AsyncResultSet.StreamMessageListener streamMessageListener);
```

Comment:
If we cannot make the interface package-private, then we should mark both the interface and this method with `@InternalApi`, and also add a Javadoc that clearly states that users should not call this method.

Reply:
we can make it as package-private